FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,16 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			switch result {
+			case let .success((data, response)):
+				let result = FeedImageMapper.getResult(from: data, response: response)
+				if self != nil {
+					completion(result)
+				}
```

Comment:
By checking if `self` has been deallocated only in the `success` path there's a possible odd behavior: `completion` might be called for the `failure` case, violating the specification that `RemoteFeedLoader`'s `load` method should not deliver a result after the instance has been deallocated.
Also, the code might be performing an expensive operation (`JSONDecoder().decode`), just to be discarded afterwards. I suggest checking for deallocation even before the switch statement on `result`.
```suggestion
				completion(result)
```